pandas/core/dtypes/cast.py

```diff
@@ -1698,7 +1742,7 @@ def construct_1d_ndarray_preserving_na(
     return subarr
 
 
-def maybe_cast_to_integer_array(arr, dtype, copy: bool = False):
+def maybe_cast_to_integer_array(arr, dtype: Union[str, np.dtype], copy: bool = False):
```

Comment:
is this limited to numpy dtypes? if not should use Dtype from typing.

Reply:
it's not. Switched to Dtype here